src/diagnostic/GraphProbe.cpp

```diff
@@ -0,0 +1,420 @@
+/** Copyright 2017-2021 CNRS-AIST JRL and CNRS-UM LIRMM */
+
+#include <tvm/diagnostic/GraphProbe.h>
+
+#include <iomanip>
+#include <iostream>
+#include <queue>
+#include <string>
+
+namespace
+{
+using namespace tvm::diagnostic;
+using namespace tvm::graph::internal;
+
+std::string indent(int val) { return std::string(val, ' '); }
+
+// shared code for any and none (see below)
+template<bool b>
+bool check(const std::vector<GraphProbe::OutputVal> & outputVal, std::function<bool(const Eigen::MatrixXd &)> select)
```

Comment:
Should also be a const reference to the function

Reply:
Fixed.